File: backend/voting-app/src/main/java/com/tu/votingapp/entities/elections/PartyEntity.java
package com.tu.votingapp.entities.elections;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "parties")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class PartyEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, updatable = false)
    private Long id;

    @Column(name = "party_name", nullable = false, unique = true)
    private String name;

    @Column(name = "abbreviation")
    private String abbreviation;

    @Column(name = "logo_url")
    private String logoUrl;

    @Column(name = "leader_name")
    private String leaderName;

    @ManyToOne
    @JoinColumn(name = "election_id")
    private ElectionEntity election;

    @OneToMany(mappedBy = "party", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<CandidateEntity> candidates = new ArrayList<>();

    public PartyEntity(Long id) {
        this.id = id;
    }
}

File: frontend/android/app/src/main/java/com/example/android/entity/election/PartyEntity.kt
package com.example.android.entity.election

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.example.android.db.orm.Column
import com.example.android.db.orm.Table

@Entity(tableName = "parties")
data class PartyEntity(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    val id: Long = 0,

    @ColumnInfo(name = "name")
    val name: String,

    @ColumnInfo(name = "abbreviation")
    val abbreviation: String? = null,

    @ColumnInfo(name = "logo_url")
    val logoUrl: String? = null,

    @ColumnInfo(name = "leader_name")
    val leaderName: String? = null
)